src/wasm-linker.cpp

```diff
@@ -387,10 +389,29 @@ bool hasI64ResultOrParam(FunctionType* ft) {
   return false;
 }
 
+void Linker::makeDummyFunction() {
+  bool create = false;
+  // Check if there are address-taken functions
+  for (auto& relocation : out.relocations) {
+    if (relocation->kind == LinkerObject::Relocation::kFunction) {
+      create = true;
+      break;
+    }
+  }
+  if (!create || out.wasm.table.names.size()) return;
```

Comment:
This function should always be called before the functions with pre-assigned indices are handled, since the pre-assigned functions will  start at index 1, so the two should be mutually compatible.  But the intention is to check just in case the table is non-empty when this function is called,  so I will convert it to an assert.